spec/controllers/lets_encrypt/verifications_controller_spec.rb

```diff
@@ -4,25 +4,39 @@
 
 RSpec.describe LetsEncrypt::VerificationsController, type: :controller do
   routes { LetsEncrypt::Engine.routes }
+  class OtherModel < LetsEncrypt::Certificate
+  end
 
   it 'returns 404 status when no valid verification path found' do
     open(:get, :show, verification_path: :invalid_path)
     expect(response.status).to eq(404)
   end
 
-  context 'has certificate' do
-    let!(:certificate) do
-      LetsEncrypt::Certificate.create(
+
+  describe 'has certificate' do
+   let!(:certificate) do
+      LetsEncrypt.certificate_model.create(
         domain: 'example.com',
         verification_path: '.well-known/acme-challenge/valid_path',
         verification_string: 'verification'
       )
     end
+    context 'with default model' do
+      it 'returns verification string when found verification path' do
+        open(:get, :show, verification_path: 'valid_path')
+        expect(response.status).to eq(200)
+        expect(response.body).to eq(certificate.verification_string)
+      end
+    end
 
-    it 'returns verification string when found verification path' do
-      open(:get, :show, verification_path: 'valid_path')
-      expect(response.status).to eq(200)
-      expect(response.body).to eq(certificate.verification_string)
+    context 'with customize model' do
```

Comment:
This test didn't verify the certificate model is changed correctly.

Reply:
You have to consider the `let!` will execute before your `context` execute `before` hook, and you still test on `LetsEncrypt::Certificate` not `OtherModel`

Reply:
Moved let!(:certificate) to each context

Reply:
It still not **verify the certificate model is changed**, you have to change some behavior the verify it and make sure the customize behavior correctly processes.